src/render3d/DisplayObjectContainerIn3D.as

```diff
@@ -26,7 +26,11 @@ package render3d {
 
 import flash.display.Sprite;
 
-public class DisplayObjectContainerIn3D extends Sprite {SCRATCH::allow3d{
+public class DisplayObjectContainerIn3D extends Sprite {
+	public static var texSizeMax:int = 2048;
```

Comment:
A while back we allowed up to 4096x4096, but that was causing a lot of noise on Sentry due to systems which didn't support it. The 2048x2048 limit was put in place by @sclements in #897. As he points out in that PR, at that time the top event was "Texture too big (max is 2048x2048)"

There doesn't appear to be any way to query Flash about maximum texture size, but we could increase texture size in a loop until we catch a texture-too-big exception.

Of course, all that is talking about 3D hardware. The change in this PR is important even when 3D acceleration isn't available. I use `texSizeMax` because it's a) it seems like a reasonable limit and b) it makes sense as a limit when we actually are running in 3D mode.